### PR TITLE
HTTP client instrumentation cleanup: google-http-client

### PR DESCRIPTION
**Pull request by trask**
Split out from #1840

### CODE REVIEW

File: instrumentation/google-http-client-1.19/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/googlehttpclient/GoogleHttpClientInstrumentationModule.java [hunk 1]
```diff
@@ -85,21 +82,19 @@ public static void methodEnter(
...
-
-      if (context == null) {
-        Context parentContext = currentContext();
-        if (tracer().shouldStartSpan(parentContext)) {
-          context = tracer().startSpan(parentContext, request, request.getHeaders());
-          contextStore.put(request, context);
-          scope = context.makeCurrent();
-        }
-      } else {
-        // span was created by GoogleHttpClientAsyncAdvice instrumentation below
```
Comment (anuraaga) [inline instrumentation/google-http-client-1.19/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/googlehttpclient/GoogleHttpClientInstrumentationModule.java:100]: I liked this comment

**Review by anuraaga — APPROVED** [review]

File: instrumentation/google-http-client-1.19/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/googlehttpclient/GoogleHttpClientInstrumentationModule.java (hunk 1 above)
Comment (trask) [reply instrumentation/google-http-client-1.19/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/googlehttpclient/GoogleHttpClientInstrumentationModule.java:100]: It's back 👍

File: instrumentation/google-http-client-1.19/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/googlehttpclient/GoogleHttpClientInstrumentationModule.java [hunk 2]
```diff
@@ -85,21 +82,20 @@ public static void methodEnter(
...
 
-      ContextStore<HttpRequest, Context> contextStore =
-          InstrumentationContext.get(HttpRequest.class, Context.class);
-      context = contextStore.get(request);
-
-      if (context == null) {
-        Context parentContext = currentContext();
-        if (tracer().shouldStartSpan(parentContext)) {
-          context = tracer().startSpan(parentContext, request, request.getHeaders());
-          contextStore.put(request, context);
```
Comment (iNikem) [inline instrumentation/google-http-client-1.19/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/googlehttpclient/GoogleHttpClientInstrumentationModule.java:96]: Is this not needed anymore?

File: instrumentation/google-http-client-1.19/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/googlehttpclient/GoogleHttpClientInstrumentationModule.java [hunk 3]
```diff
@@ -138,15 +125,12 @@ public static void methodEnter(
         return;
       }
 
-      context = tracer().startSpan(parentContext, request, request.getHeaders());
+      context = tracer().startSpan(parentContext, request);
+      scope = context.makeCurrent();
 
       // propagating the context manually here so this instrumentation will work with and without
       // the executors instrumentation
```
Comment (iNikem) [inline instrumentation/google-http-client-1.19/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/googlehttpclient/GoogleHttpClientInstrumentationModule.java:132]: I find this comment suspicious. Why of all places here to want to work without executors instrumentations?

File: instrumentation/google-http-client-1.19/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/googlehttpclient/GoogleHttpClientInstrumentationModule.java [hunk 4]
```diff
@@ -113,16 +109,7 @@ public static void methodExit(
...
       scope.close();
-      if (throwable == null) {
-        tracer().end(context, response);
-      } else {
-        tracer().endExceptionally(context, response, throwable);
-      }
-      // If HttpRequest.setThrowExceptionOnExecuteError is set to false, there are no exceptions
-      // for a failed request.  Thus, check the response code
-      if (response != null && !response.isSuccessStatusCode()) {
-        spanFromContext(context).setStatus(StatusCode.ERROR);
```
Comment (iNikem) [inline instrumentation/google-http-client-1.19/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/googlehttpclient/GoogleHttpClientInstrumentationModule.java:124]: Is this not needed anymore?

File: instrumentation/google-http-client-1.19/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/googlehttpclient/GoogleHttpClientInstrumentationModule.java (hunk 3 above)
Comment (trask) [reply instrumentation/google-http-client-1.19/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/googlehttpclient/GoogleHttpClientInstrumentationModule.java]: I agree this comment is suspicious, so I removed it 😁.

I think we always propagate the span out-of-band to the place where it is ended. (this was actually something nice about the `Operation` concept, b/c you had to pass the Operation object out-of-band to the place where it was ended, orthogonal to in-band Context propagation)

**Review by iNikem — APPROVED** [review]

**Pull request merged**